packages/design-system/src/components/client/toggle.tsx

```diff
@@ -0,0 +1,14 @@
+import { Switch as SwitchHeadless, type SwitchProps as SwitchPropsHeadless } from "@headlessui/react";
+import * as React from "react";
+
+// biome-ignore lint/suspicious/noExplicitAny: allow any for generic `as` prop support
+export type Toggle = SwitchPropsHeadless<any>;
```

Comment:
Mame nejaky use case pro `Toggle`, co neni nas `Button`? Potom by bylo asi dobre vyresit ho zvlast a ne genericky.

Protoze tady se da ten typ dobre zafixovat jako:

```
export type Toggle = Omit<SwitchPropsHeadless<typeof Button>, "as">;
```

Stejne jako je zafixovane as u samotneho Buttonu. Zmizi `any` a zbytecne rozhodovani jaka komponenta se ma pouzit v kodu.

Zafixuje se to primo tu:
```
<SwitchHeadless {...props} as={Button} ref={ref} />;
}
```

Reply:
Probably not now, at least not anything that will not be part of the design system so that we can't hardcode it, but this approach will block you from using this for components outside the design system (which can happen).